quadriga/src/main/java/edu/asu/diging/quadriga/config/MongoConfig.java

```diff
@@ -68,25 +73,34 @@ public MongoDatabaseFactory mongoDbFactory() throws UnknownHostException {
     }
 
     @Bean
-    public MongoTemplate mongoTemplate() throws UnknownHostException {
-        return new MongoTemplate(mongoDbFactory());
+    public MongoTemplate mongoTemplate(@Qualifier("mongoDbFactory") MongoDatabaseFactory mongoDbFactory) throws UnknownHostException {
+        MappingMongoConverter converter = new MappingMongoConverter(
+                new DefaultDbRefResolver(mongoDbFactory), new MongoMappingContext());
+        converter.setCustomConversions(customConversions());
+        converter.afterPropertiesSet();
+        return new MongoTemplate(mongoDbFactory, converter);
     }
     
     @WritingConverter
-    @Component
-    class OffsetDateTimeWriteConverter implements Converter<OffsetDateTime, String> {
-        @Override
+    enum OffsetDateTimeWriteConverter implements Converter<OffsetDateTime, String> {
+        INSTANCE;
         public String convert(OffsetDateTime source) {
             return source.toInstant().atZone(ZoneOffset.UTC).toString();
         }
     }
 
     @ReadingConverter
-    @Component
-    class OffsetDateTimeReadConverter implements Converter<String, OffsetDateTime> {
-        @Override
+    enum OffsetDateTimeReadConverter implements Converter<String, OffsetDateTime> {
+        INSTANCE;
         public OffsetDateTime convert(String source) {
             return OffsetDateTime.parse(source);
         }
     }
```

Comment:
why are the above two classes turned into enums?

Reply:
Most of the default mongoConverters are enums. org.springframework.data.mongodb.core.convert.MongoConverters